tests/functional/verification/test_reflected_,magic_func.py

```diff
@@ -0,0 +1,202 @@
+from nagini_contracts.contracts import *
+
+############################################
+##  Basic tests                           ##
+############################################
+
+class A():
+  def __add__(self, other: 'B') -> 'A':
+    return A()
+
+class B():
+  pass
+  
+r1 = A() + B()
+
+
+# __add__ of left operand takes priority
+class C():
+  def __add__(self, other: 'D') -> 'C':
+    return C()
+  
+class D():
+  def __add__(self, other: 'C') -> 'D':
+    Requires(False)
+    assert False
+  
+r2 = C() + D()
+
+
+# __add__ prefered over __radd__
+class E():
+  def __add__(self, other: 'E') -> 'E':
+    return E()
+  
+  def __radd__(self, other: 'E') -> 'E':
+    Requires(False)
+    assert False
+  
+r3 = E() + E()
```

Comment:
Like for the other PR, it would be great if you could also add a few statements that should lead to verification errors. Since you'll only get at most one error per function and one error for all top-level statements in Silicon, that might require putting some of the top-level statements here into separate functions.